src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -238,12 +238,7 @@ public async Task StopAsync(CancellationToken cancellationToken)
 
         public async Task RestartHostAsync(CancellationToken cancellationToken)
         {
-            if (Interlocked.CompareExchange(ref _restarting, 1, 0) != 0)
-            {
-                // only one restart operation at a time
-                return;
-            }
-
+            await _semaphore.WaitAsync();
```

Comment:
One additional nit... can this have a more descriptive name than just `_semaphore`?